vc-worker.py

```diff
@@ -161,36 +211,13 @@ def action_others(conn, action, vc):
         elif request_type == 'stop':
             action_result = action_stop(request, machine_ro, vc)
         elif request_type == 'get_info':
-            logger.debug(request.to_dict())
-            try:
-                info = vc.get_machine_info(machine_ro.provider_id)
-                logger.debug(info)
-            except Exception:
-                logger.error('get_info exception: ', exc_info=True)
-                info = {'ip_addresses': [], 'nos_id': '', 'machine_search_link': ''}
-
-            machine = data.Machine.get_one_for_update({'_id': request.machine}, conn=conn)
-            machine.nos_id = info['nos_id']
-            machine.machine_search_link = info['machine_search_link']
-
-            if len(info['ip_addresses']) != 0:
-                machine.ip_addresses = info['ip_addresses']
-                machine.save(conn=conn)
-                request.state = 'success'
-                action.lock = -1
-            else:
-                machine.save(conn=conn)
-                request.state = 'delayed'
-                action.repetitions -= 1
-                action.next_try = datetime.datetime.now() + datetime.timedelta(
-                    seconds=random.randint(action.delay, action.delay+3)
-                )
-                action.lock = 1
-            request.save(conn=conn)
-            action.save(conn=conn)
+            action_get_info(request, machine_ro, vc, action, conn)
```

Comment:
If I understand it correctly, this action does not override `action_result` and therefore will result in setting `failed` state to `machine.state`.

Related to line 206.

Reply:
it ends up with `return` and all required settings of action/request/etc. is done in the function